### ISSUE TITLE
Pagination

### ISSUE THREAD

**Issue by govuk-design-system** [alphagov/govuk-design-system-backlog#77]
Use this issue to discuss the [Pagination component in the GOV.UK Design System](https://design-system.service.gov.uk/components/pagination/).

## What
Help users navigate through multiple pages. Typically used for pages of search results.

## Examples

Multi-page pagination examples:

- https://design-patterns.service.justice.gov.uk/components/pagination/
- http://hmcts-design-system.herokuapp.com/components/pagination
- https://design.homeoffice.gov.uk/components?name=Pagination
- https://cjscpp-design-patterns.herokuapp.com/design_patterns/pagination
- https://dvsa-front-end.herokuapp.com/library/mts/pagination

Previous / next only pagination examples:

- http://land-registry-elements.herokuapp.com/pagination-simple/demo
- https://digital-land-design.herokuapp.com/component/pagination
- https://nhsuk.github.io/nhsuk-frontend/components/pagination/index.html
- https://components.publishing.service.gov.uk/component-guide/previous_and_next_navigation

**Comment by joelanman** [comment]
[Pagination in the US Web Design System](https://github.com/uswds/uswds/issues/959)

**Comment by 36degrees** [comment]
There was [a request for this in Elements](https://github.com/alphagov/govuk_elements/issues/538).

**Comment by gazaston** [comment]
Pagination from [blog.gov.uk](blog.gov.uk):

<img width="1116" alt="screen shot 2018-09-25 at 10 27 19" src="https://user-images.githubusercontent.com/822507/46005644-a088df80-c0ad-11e8-97c9-b423b15b5ad8.png">

**Comment by glenjamin** [comment]
Are there scenarios where "jump to page N" is actually desirable to a user?

In practice I can see why next/previous is useful in a long list, but jumping to a numbered page is probably always going to be worse than providing the user a way to filter down the results to something manageable.

**Comment by 36degrees** [comment]
One scenario I can think of is a user 'binary searching' a relatively small number of sorted results. For example, with an imagined alphabetical result set spanning 10 pages, if a user is looking for something beginning with 'N' they might jump to page 5 (guessing, based on the position of n in the alphabet) and then navigate from there to find the result they're looking for, which would _arguably_ be a better experience than having to page through one page at a time.

I'd agree that providing a search or filtering interface would be preferred, but _might_ not always be possible?

**Comment by mgilgar** [comment]
I think 90% of the times it should be enough to provide only the first X results without pagination and give the option to filter or seach. Less complicated for the user and for the developer. 

Anyway, I can see the needs of the other options for some scenarios. I think it would be great if govuk can develop patterns for all of them.

Also, please specify what the url should look like.

**Comment by edwardhorsford** [comment]
Came here as my service uses / needs pagination.

One use case to consider:
Where the user isn't looking for a specific thing (so search / filtering isn't necessarily relevant), but wants to review *all the things*. For instance, my user might want to review all the things created by *Ed*. They may filter for _Ed_ as creator, and then want to read down the entire list of things _Ed_ created.

On a personal level (and very much subjective to how I work), having numbered pages may help if people are noting down _things of interest_ - I can review several pages and know that my interested things are on pages 1 and 3.

Numbering potentially gives an idea of how many pages of results there are - but there are other ways this could be achieved.

**Comment by tomwrightgov** [comment]
Wow, there is a lot of different styles in use out there in .gov sites. 

My personal preference is for the ‘blog.gov.uk’ style.  Labelling the links ‘Previous page’ and ‘Next  Page’ is a lot more obvious then ‘Prev’ and ‘Next’ especially for screen readers.
Listing links to the surrounding page numbers with additional links to the first and last page is also a nice touch. 
My guess is that this component can straight from the blogging engine and so is widely used outside .gov.

The thing I am most uncertain of is the use of item counts and pages numbers. 

Is it better to say:

Page 3

or

Showing 21 – 30 of 100 items

Does the user care how many items there are?

Is it enough to just show the number of pages, as that it the quantity that the user is actually dealing with.

DO we show both number of items and pages, like land-registry does. 
Does that add any value? 
Is it confusing to the user?

**Comment by leekowalkowski-hmrc** [comment]
Well, the word 'items' isn't a geat choice of content, so I don't know if the user would care.  In the Government Gateway's group management, we have 'Showing 21-30 of 100 team members' and I think yes, a group administrator could care how many team members were in the group (or matched their search criteria).

**Comment by mtallamy** [comment]
I thought I'd offer some feedback we received during private beta for our service. Our page closely follows the first example given at the top of this page, i.e. https://home-office-digital-patterns.herokuapp.com/components/pagination. In addition to further filtering, we've been asked by some of our users to add 'first' and 'last' page links alongside the 'previous' and 'next' links to help traversing larger data sets.

**Comment by roobottom** [comment]
[This article](https://www.a11ymatters.com/pattern/pagination/) is useful from *Accessibility Matters*.

**Comment by edwardhorsford** [comment]
We’ve just added pagination to our service - which is a modified form of the [MOJ pagination](https://moj-design-system.herokuapp.com/components/pagination). [Our PR adding it](https://github.com/DFE-Digital/register-trainee-teachers/pull/517).

What it looks like:
![Screenshot 2021-02-15 at 15 06 48](https://user-images.githubusercontent.com/2204224/107963039-92e69400-6f9f-11eb-8f1d-071342de7b72.png)

Changes from the MOJ one:
* Add `aria-label` to the `nav`, removing the paragraph.
* Include the page number `(page x of y)` in the page title _if_ pagination is visible.

Some other implementation details:
* The pagination is all driven from the query string - so `&page=4` will take you to page 4.
* If you attempt to load a page out of bounds, we redirect you to page 1.
* Any change to filters on the page will reset the pagination.
* We're starting with a page size of 100, but will evaluate this in the future and whether we need it configurable (I hope we don't).

**Comment by terrysimpson99** [comment]
See: https://gouvfr.atlassian.net/wiki/spaces/DB/pages/223051980/Pagination+-+Pagination

**Comment by hopebristow** [comment]
We have used pagination in 3 places in our service for check and pay for multiple vehicles as part of the "Drive in a Clean Air Zone" service https://www.gov.uk/clean-air-zones. We also adapted a version of the MoJ pagination https://moj-design-system.herokuapp.com/components/pagination. Originally, we used a simple “next” and “previous” in one of these designs but users found this confusing and weren’t sure where they were in their overall list of vehicles when asked to find and make payments from a list of number plates that spanned multiple pages, often resulting in them failing to complete the task. When we changed this to match the MoJ pattern we saw an increase in the usability for users and all were successful in completing the task. All 105 users we tested with had no issues when testing this page with page numbers added. 

![image](https://user-images.githubusercontent.com/39621477/116227484-1e249880-a74c-11eb-9e7e-9d0be2100dd6.png)

**Comment by calvin-lau-sig7** [comment]
### GOV.UK Design System working group review

The Design System working group reviewed this contribution in November and December 2021. They’ve [approved it to be usable, consistent and versatile.](https://design-system.service.gov.uk/community/contribution-criteria/)

To get it ready to publish, the working group made these recommendations.

On the scope of this component, they made recommendations to:

- focus on ‘numbered pagination’, which is the most useful implementation for service teams
- split out and possibly remove the ‘previous or next’ variant, as it’s not strictly about pagination
- include support for the user to navigate to any page (such as with a number input) — however, the working group agreed this was ”‘one for the future”

On the design of this component, they made recommendations to:

- consider if the component needs to include a top border
- consider if users need the guidance and example to ‘show the total number of results’
- consider if the mobile version should be centre aligned — we said that this made sense as the component is symmetrical and nearly the width of a mobile device screen, similar to how full-width buttons appear
- make arrows and chevrons more consistent — we said that ideally, arrows should represent movement and chevrons should represent dividers, which this component works towards
- make links a consistent colour, as some are blue and some are black — we said this was to be consistent with a visual hierarchy formed around the ‘back link’ and ‘breadcrumb’ components, though we do need to document these design principles
- consider if users need the guidance and example for ‘showing large numbers of pages’

On the guidance, they made recommendations to:

- explain why the previous and text links are stacked on top of each other — we’ll add guidance to explain that this makes the next link more visible, especially for users of screen magnifiers
- clarify if services should ever place this component before the pages — however, we do not know if there is a reasonable use case for this
- suggest services also look at using search filters to help improve navigation

On the coded examples, they made recommendations to:

- include an example of paginating specific parts of page, such as a table — however, this might not be practical, as this could only be done with JavaScript
- remove an extra border from the full page example — but this will depend on whether we decide to keep the top border
- set the arrow size in rems, so it scales properly
- consider using `divider: true` rather than `type: ellipsis`
- unquote the Nunjucks keys

**Comment by adamsilver** [comment]
The pagination component cannot be configured to show a line for ‘Showing 10 to 20 of 256 results’.

![image](https://user-images.githubusercontent.com/37163/179248772-f9cece52-243f-407e-8abe-df5a50baba7f.png)

This leaves designers and developers 3 options that I can think of:

(1) Do not give users that information

The problem with this is that users don't get this information.

(2) Show the information somewhere "outside" of the bounds of the pagination component - for example, underneath as a paragraph

![image](https://user-images.githubusercontent.com/37163/179249011-7237a52f-0e0b-46b9-a1dc-b8245123a32d.png)

The problem with this is that I still have to implement this layout myself and it's likely to mean that lots of different services implement the same thing inconsistently.

(3) Create my fork of the pagination component to include this capability

The problem with this is the amount of effort it will take to do this plus it means not using the design system component directly.

I have not seen users struggle or complain without a line for "Showing...". But:

- It is something I've seen across many different services across many departments
- It's been provided in departmental design systems such as [HMCTS](https://hmcts-design-system.herokuapp.com/components/pagination), [MOJ](https://design-patterns.service.justice.gov.uk/components/pagination/), [HO](https://design.homeoffice.gov.uk/components?name=Pagination)

**Comment by armstrongb** [comment]
Also in the [ONS design system](https://ons-design-system.netlify.app/components/pagination/).

**Comment by simonwhatley** [comment]
To echo @adamsilver's comments, I think that "Showing X to Y of Z results" is a fundamental part of pagination to help orientate the user.

Alongside the previously mentioned variants used at DfE, MOJ, HMCTS and the Home Office, [Hackney](https://design-system.hackney.gov.uk/components/pagination) takes the same approach (also allowing its removal), and [ONS](https://ons-design-system.netlify.app/components/pagination/) has a design variant.

For balance, it's worth noting Digital Scotland and the MOD design systems only show page numbers, DWP and HMRC don't appear to have a pagination component, and the NHS use previous/next.

The point "consider if users need the guidance and example to 'show the total number of results'" seems to have been taken to mean that "Showing X to Y of Z results" is not essential for the component. I don't think this was the correct interpretation of the working group's feedback.

**Comment by simevidas** [comment]
Quick question: Would it make sense to remove the `href` attribute from the `aria-current=page` link? I just find it a bit odd that it’s possible to click and focus the current page link. In these pagination components, I prefer when the current page link is not interactive.

<img width="601" alt="Screenshot 2022-08-13 at 10 07 23" src="https://user-images.githubusercontent.com/716405/184475171-481feae4-c957-4cbf-a762-bebfa20a4456.png">

**Comment by Katecheshire1999** [comment]
Does anyone know why my pagination doesn't seem to be linking to any styling? I copied the HTML straight from the pattern and all the other gov uk styles are working on the same page!

<img width="223" alt="Screenshot 2022-08-15 at 16 34 08" src="https://user-images.githubusercontent.com/90384860/184666328-67ad91c7-0b81-44dc-bf24-2e79afbce8c2.png">

**Comment by querkmachine** [comment]
@simevidas You should include the `href` if using an `<a>` element, otherwise you should use a different element (like a `<span>`) instead. `aria-current` is also most effective if used on a link, as the purpose of the attribute is to provide context about which link in the group the user is currently viewing.

@Katecheshire1999 Are you using version 4.2.0 of govuk-frontend (or newer)? The pagination component wasn't present in older versions. If you're using the Prototype Kit, it hasn't updated to include the pagination component just yet.

**Comment by Beccawill334** [comment]
@Katecheshire1999 Did you find a fix for this? Mine is doing the same thing even though I have the latest version of the prototype kit!

**Comment by querkmachine** [comment]
@Beccawill334 The Prototype Kit hasn't been updated to include the pagination component yet. A release that includes it is expected to be released next week.

**Comment by itsmestevey** [comment]
We are looking to introduce pagination on Manage Incentives in HMPPS at the MoJ. Does anyone know whether there are any recommendations on how many items could/should appear before pagination takes place?

**Comment by CharlotteDowns** [comment]
We ran an external accessibility audit for some of the components and patterns in GOV.UK Frontend in May 2023. In that audit, we included an example of the Pagination component. We’re adding results from that audit here so that we can:
- document, discuss and address the findings
- inform future contributors of the findings

## Three accessibility issues raised

### Issue 1: WCAG AAA

The 'next' and 'previous' labels could use a bit more context. 'Next page'?
> The ‘Next’ and ‘Previous’ links in the pagination do not possess additional link text of ‘page’ or similar to inform users of screen reading assistive technologies of their purpose when viewing the links out of context using the link text alone.

More detail in this issue:
- [#3679](https://github.com/alphagov/govuk-frontend/issues/3679)

### Issue 2 : Usability

We don't show the current page number / title in the pagination component. Users gotta guess!
> The pagination method used does not indicate to users which page they are currently on and relies on the user to decipher this information from the context of the previous and next links, which some users may not be able to do with ease.

More detail in this issue:
- [#3682](https://github.com/alphagov/govuk-frontend/issues/3682)

### Issue 3 : Usability

Poor region label on the pagination for small # of pages. Change to something more descriptive.
> The pagination region has been marked up within a element. However, this has been provided with an accessible name of ‘results’. This does not inform the user of the purpose of the region and does not convey that it contains the pagination options. As screen reader users will often navigate via regions, it is important that they are labelled adequately.

More detail in this issue:
- [#3684](https://github.com/alphagov/govuk-frontend/issues/3684)

**Comment by GraemeLaurenson** [comment]
Hi all - I’m Graeme, I’m currently an ID with [RoS.gov.uk](http://ros.gov.uk/)  .. hope this is the right place to front this - if not .. my apologies and please point me in the right direction ..

We currently have an issue (and proposals) relating to using the pagination component in our Sasines Search tool.
The benefits of this component are that it lets the user see the first and last pages, and the previous and next from the current page.
However the default component is too wide - potentially over 600px.
It has the duplication of 2 controls, the ‘Previous’ and ‘Next’ have the same effect as the numbers either side of the current page number (blue box)
Proposal 1: We’d like to remove the words ‘Previous’ and ‘Next’ - still allowing the user to go to those pages - but saving 118px. 
Proposal 2: We’d like to remove the words ‘Previous’ and ‘Next’ as above and tighten up the spacing - but now saving 206px!
We have user-tested with no issues.

The attached image demonstrates the usage/benefits.

Please can we have comments/advice with this - with the potential to propose a change through GitHub?

Thanks
G.
![Pagination proposals - Sep 23](https://github.com/alphagov/govuk-design-system-backlog/assets/144107906/2d00abe5-0675-46f7-911b-436aec11c096)

**Comment by owenatgov** [comment]
Hi @GraemeLaurenson, thanks for the contribution.

Whilst we appreciate the thorough outline of your proposal, we'd like to understand the need for a less wide pagination component. Going over the suggestions you made and addressing why they are the way they are:

- The explicitly spelled out navigation labels make the component more usable as they indicate what the link is going to do in plain language. There is a slight but valid cognitive load to relying more on the number links next to the current page. It also makes the component easier to use and digest by screen reader users.
- The spacing is principally to address WCAG requirements for minimum touch target size. [The AAA criterion states that a touch area should be 44px by 44px](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html), but we feel there's no harm in extra spacing. Having said that, you have identified that the padding on the ellipses could collapse against the numbers and save a little extra real estate.

**Comment by GraemeLaurenson** [comment]
Hi Owen - thanks for the response.

We have limited space to use the component - and while other (legacy) components would fit, they don’t offer the functionality.
The users are all internal colleagues who will be trained on the journey and therefore there’s a factor of ‘learned behaviour’.
We also have some preliminary research and users had no issues.

I've noted that the Digital Scotland Design System hides the ‘previous’ and ‘next’ text in smaller viewports.
https://designsystem.gov.scot/components/pagination/middle-example

Which looks much tidier than..
https://design-system.service.gov.uk/components/pagination/for-list-pages/index.html
(Try them at <768px window)

Is there a way an improved ’small viewport’ version of the GDS component can be built? It seems it would be useful for others too! 
I’m assuming however there are a lot of hoops to jump through?!

Interested in your thoughts
Thanks
G.

**Comment by owenatgov** [comment]
Hi @GraemeLaurenson. Thanks for the clarification.

In this case, if this setup works for your users then I would recommend simply changing the html of the component. You can do this by removing the `span` with class `govuk-pagination__link-title` from within the prev and next links. This will remove the text label. I would recommend setting an accessible name on the `a` tags that contain the prev and next links so that they still work for screen readers eg: adding the attribute `aria-label="previous page"` to the previous arrow icon link.

We are unlikely to implement this and deliver it via GOV.UK Frontend for the reasons outlined in my previous comment and our research that supports these positions, and because adding a solution for a specific use case creates additional weight on the component that could negatively benefit public facing services using this component. If you however are confident that this amendment works for your users then you can amend the component to suit your service.

**Comment by nasainsbury** [comment]
At work, we have a simple pagination component which has a `Previous` and `Next` button.

There is some debate as to how these buttons should be styled. Currently the designers are proposing that `Previous` be a "Secondary" button, whilst `Next` is "Primary. I.e

<img width="1512" alt="Screenshot 2024-03-04 at 10 45 00" src="https://github.com/alphagov/govuk-design-system-backlog/assets/58425475/d27570f0-1b29-46ce-a32f-d5dffbe5df4b">

I am of the view that both these buttons do the same "action" (in essence) and are of equal "importance", hence I'd have them styled the same.

Curious on thoughts around this when no number is included with the pagination component. TIA.

**Comment by joelanman** [comment]
@nasainsbury The GOV.UK Pagination component is here, including a numberless variant:
https://design-system.service.gov.uk/components/pagination/
It uses links not buttons, and does not emphasise previous or next

**Comment by nathan-at-tes** [comment]
@joelanman I checked this component before coming here, the differences I spotted were:

- In the GDS example without numbering, the pages seemed to be different content.
- The example I provided is a semantic button as it doesn't actually load a new page, it just does some JavaScript to "reload" the content. Although, there's no reason it cannot be a new link as we're using Next.

This may actually be opening up a second question then, in our pagination example we do not allow users to select a page they want to go to, only the next or previous (if available). This was due to technical constraints in the API. We also do not show the current "page".

Since the "previous"/"next" buttons are not taking us to completely different content, (as I'm assuming is happening in the picture below), would this still be the right component to use, but without secondary text below each?

![image](https://github.com/alphagov/govuk-design-system-backlog/assets/111738668/d9640ef3-7339-465b-bea1-42e6227ed298)

Also, the comments about secondary/primary buttons is that our designers are putting focus on going "next". You've said the component "does not emphasise previous or next", is that a conscious decision and/or is there a blog post from GDS explaining the rationale behind that? I'm rather keen to follow this logic, but have nothing other than a "gut-feeling".

**Comment by nasainsbury** [comment]
@joelanman , prior to our discussion, I reviewed this component and noted the following variances:

In the GDS example without numbering, the pages appeared to contain distinct content. Conversely, the example I presented showcases a semantic button. It executes JavaScript to "reload" content without necessarily directing to a new page. However, considering our utilization of Next.js, there's potential for it to function as a new link.

This observation potentially leads to another inquiry: In our pagination model, users are restricted from selecting a specific page to navigate to, only afforded the option of moving to the next or previous page, if available. This limitation stems from technical constraints within the API. Furthermore, we omit displaying the current "page".

Given that the "previous" and "next" buttons don't lead to entirely disparate content (as depicted in the image below), would this component still be appropriate for our use, albeit without supplementary text beneath each button?

![310058229-d9640ef3-7339-465b-bea1-42e6227ed298](https://github.com/alphagov/govuk-design-system-backlog/assets/58425475/cf601de2-eecc-4e87-9322-d3fc9fc33170)

Furthermore, regarding the discourse on secondary and primary buttons, our designers are prioritizing the "next" action. Your remark that the component "does not emphasize previous or next" prompts curiosity about whether this stance is intentional and if there exists a GDS blog post elucidating the rationale behind it. I'm eager to align with this reasoning, albeit currently lacking concrete justification beyond intuition.

**Comment by joelanman** [comment]
on using a javascript framework - we do have a requirement to use progressive enhancement:

https://www.gov.uk/service-manual/technology/using-progressive-enhancement

if you are updating the page using javascript you would have to make sure it is progressively enhanced, and that the javascript interaction is accessible for example to screen reader users. If that's the case, its possible you need gov uk secondary buttons: https://design-system.service.gov.uk/components/button/

We only use primary green buttons to highlight the main call to action on a page

**Comment by nasainsbury** [comment]
putting aside the PWA, and whether it's a button/link, the primary concern is around the styling of the "previous" and "next" link.

all examples I find online, the previous/next links are styled to look the same. in the example I've presented, we can assume it behaves in a non-framework style, and when you click next, it goes to website.com/search?page=3, and when clicking previous it goes to website.com/search?page=1.

we've followed the guidance on removing the button when you cannot go back or forward.

the main question is, would there be any reason to style the next/previous button differently? I can so no examples where this is the case, and in all the GDS examples they are styled the same. is there a specific reason for this?

apologies if I'm not explaining this well enough.

TLDR; Should the previous and next links be styled to look the same in all cases?

**Comment by joelanman** [comment]
We only use a different design (primary button) when it is a main call to action on the page (for example submitting a form). In all other cases the normal link or secondary button style is used, or the design of a specific component if it exists like pagination, tabs and so on. If you do user research and find another style works better you can feed that back in this thread.

**Comment by nasainsbury** [comment]
OK perfect. Thank you. I am also curious to see if we get any feedback about this, as the logic presented to me was that _we want people to view more and thus wanted to give more attention to **next** than **previous**_

appreciate the help!

**Comment by robertjmccarthy** [comment]
The MoJ Design System has a [pagination component](https://design-patterns.service.justice.gov.uk/components/pagination/) which we're currently reviewing. We think this is a candidate for archiving from the MoJ Design System and signposting to the [GOV.UK Design System's pagination component](https://design-system.service.gov.uk/components/pagination/) instead.

But, as seen in this discussion thread, and in discussion with teams at MoJ, the GOV pagination component does not include a way to show the number of results alongside the pagination navigation.

The MoJ Design System team think there are 3 options that we could consider:

**Option 1**
The MoJ pagination component is archived and we contribute to this discussion for a need to show results on the GOV pagination component.

**Option 2**
As above in option 1, but instead of contributing to this discussion, the MoJ Design System team submits a PR for review. Once reviewed and merged, the MoJ pagination component would be archived.

**Option 3**
The MoJ Design System archives the pagination component, but creates a 'Show results' component (name firmly TBC) which references the GOV pagination component to be used as a complimentary and connected component.

The MoJ Design System teams preference would be for option 2. Option 1 leaves users with a choice of using an unsupported component from the MoJ Design System, or a component from the GOV.UK Design System that doesn't do all they need. Option 3 feels far from ideal due to the work required to implement it.

Could the GOV.UK Design System team let us know whether a 'show results' is on the table for the pagination component, and if not, would a PR be welcomed?

**Comment by querkmachine** [comment]
(Personal thoughts, not responding on behalf of the team.) 

I personally don't see labels like "96 results" or "Showing results 1–10 of 96" to be necessary to include within a pagination component. It's related information, sure, but it's also not information I really expect to be in the same place as pagination — rather, I expect it to be nearer to the top of the page. 

Maybe this is just an expectation created because that's where I tend to see it or notice it, but it doesn't seem to be uncommon to position it that way.

Bing shows the total number of results at the top of the page, below the search form and filters and above the first result. Google no longer shows this information at all, but when it did, it did so in a similar location.

<img width="403" alt="Screenshot from Microsoft Bing search results page for 'Windows 95', with the search field, a list of filters by type, and text reading 'About 11,700,000 results'." src="https://github.com/user-attachments/assets/9a6a51e5-98e0-492b-84fd-71f879b99aa6">

GOV.UK search, likewise, shows the total number of results at the top.

<img width="669" alt="Screenshot of a GOV.UK search for 'Windows 95', showing a search field, a toggle to show and hide filters, and small text reading '139 results'." src="https://github.com/user-attachments/assets/7092eb2f-a9f5-4c0a-b213-009bf84f50d0">

<hr>

Some other examples of major websites showing the total number of results separate from pagination and above the results themselves.

<img width="370" alt="Screenshot of an Amazon search for 'Windows 95', showing a search field, various navigation elements, and text reading '1-16 of over 7,000 results for windows 95'." src="https://github.com/user-attachments/assets/08404297-949e-4f8a-aa95-112aeb12e1c6">

<img width="946" alt="Screenshot of Wikipedia's search results page, with search field, filters and 'Results 1 – 20 of 19,766' shown above any results." src="https://github.com/user-attachments/assets/b483bfa6-3844-48bf-8b0a-ec9d45315cfc">

<img width="403" alt="Screenshot of a search on Skyscanner, of a much busier interface with lots of filter and sorting options, but still showing '1109 results' above the list of results." src="https://github.com/user-attachments/assets/56ed1073-6bba-40e4-9ad7-bd08bc09d17c">

**Comment by Ciandelle** [comment]
Hi Rob, 

The team had initially designed the pagination component with the total number of results in it, however when we reviewed it we decided to remove it for the following reasons:

- We felt that this guidance would be more at home in a ‘search results’ pattern
- By including it in the component, we’d be taking away the freedom to put this information wherever teams want to put it – it doesn’t necessarily make sense for it to be at the bottom of the page etc
- Even if we remove this, we should advise users not to have 100s pages of results unless they have good search and filters
- It’s dependent on the number of results you get back – you don’t need it if the total number is only a page

**Comment by robertjmccarthy** [comment]
Thanks @Ciandelle. Rationale all makes perfect sense. Any plans to make a 'search results' pattern? I think that covers the **Option 3** that I mentioned which is the route we'd likely go down unless there's plans for your team to be working on one.

**Comment by Ciandelle** [comment]
@robertjmccarthy Just checked with the team and it's not on any roadmap so it doesn't look like we'd be looking into it any time soon. I'd love to see what you're going to do with it though!

**Comment by PatriciaSanchezHMRC** [comment]
Hello all! 

Just jumping in to share a few insights from our recent user research with screen reader users, which touched on this variation of the pagination component (the one with the ellipsis) - see [**'For navigating pages of items'** on GOV.UK](https://design-system.service.gov.uk/components/pagination/).

<img width="532" height="275" alt="Image" src="https://github.com/user-attachments/assets/8f01e0e1-08c4-44ef-8463-ac90ba088668" />

We use this pagination variation in cases with many transactions and pages, where showing all page numbers would be overwhelming. 

The main issues we found were:

- When the user focused on the `...`, their screen reader announced it as “horizontal ellipsis”. We were curious if that’s the expected behaviour, or should we be providing a more descriptive name using an `aria-label`.

- The user also raised a question about the missing pages between 3 and 17. They couldn’t tell if those pages existed.

- Finally, they expected to be able to jump to a specific page number manually, but since that wasn’t supported, they had to keep pressing ‘Next’ to get where they needed, which was pretty frustrating for them.

Hope the research offers helpful insights.

**Comment by querkmachine** [comment]
@PatriciaSanchezHMRC 

> * Finally, they expected to be able to jump to a specific page number manually, but since that wasn’t supported, they had to keep pressing ‘Next’ to get where they needed, which was pretty frustrating for them.

I find that piece of feedback interesting, because how would a user know which of a set of sequential pages the thing they're looking for is on? Even if the ability to jump to a specific page number were available, it would still require moving back and forth and making repeated lookups to find a specific result (unless they were lucky and got it right on the first guess)

It feels like solutionising without stating the actual problem: that they expect the results to be searchable, filterable, or grouped in a different way to how they currently are.

**Comment by PatriciaSanchezHMRC** [comment]
@querkmachine 

Hey there!

That’s a great point. You’re right that that the request to jump to a specific page highlights a broader need for improved filtering and navigation of large result sets. It’s something we've picked up on in our research too.

I still thought it might be useful to share the user feedback we’ve seen around this. Thanks for your input!

**Comment by gavinthomas-valtech** [comment]
I notice in the example it has a bare # used in the href `href="#" `. Is this not bad practice? It's linking to the same page? should it not be a link since it's linked to itself?

How will a screenreader read this out? it will reload the same page they are on?

**Comment by 36degrees** [comment]
@gavinthomas-valtech the links used in the examples are just placeholders, for two reasons:

1. We don't know what URL structure will be needed, as it'll be unique to the service.
2. If we used real links, clicking them would attempt to load the URL in the example iframe (likely resulting in a 404).
 
You can see we do the same thing with `#` for links elsewhere, for example with the examples for the breadcrumbs component.

**Comment by gavinthomas-valtech** [comment]
I guess what i'm asking is what is the pattern for a link that is the same page when used in a pagination component? It uses # as an example. I would have thought this should not be an actual link at all as it links to the same page? or do we actually use a link to eh same page as use aria-current to define its the current page?

> [@gavinthomas-valtech](https://github.com/gavinthomas-valtech) the links used in the examples are just placeholders, for two reasons:
> 
>     1. We don't know what URL structure will be needed, as it'll be unique to the service.
> 
>     2. If we used real links, clicking them would attempt to load the URL in the example iframe (likely resulting in a 404).
> 
> 
> You can see we do the same thing with `#` for links elsewhere, for example with the examples for the breadcrumbs component.
